wire-schema/src/main/java/com/squareup/wire/schema/WireType.java

```diff
@@ -0,0 +1,135 @@
+package com.squareup.wire.schema;
+
+import com.google.common.base.Joiner;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.squareup.wire.internal.Util;
+import java.util.Objects;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.base.Preconditions.checkState;
+
+/**
+ * Names a protocol buffer message, enumerated type, service or a scalar. This class models a
+ * fully-qualified name using the protocol buffer package.
+ */
+public final class WireType {
```

Comment:
Is this a `ProtoType`? For some reason I have pause about using "Wire" in the names (I feel the same about the things in the runtime, too).